shops/code/controllers/shopController.js

```diff
@@ -0,0 +1,133 @@
+import { JSONFilePreset } from "lowdb/node";
+
+// Read or create db.json
+// defaultData specifies the structure of the database
+const defaultData = { meta: {"tile": "List of shops","date": "November 2024"}, shops : [] }
+const db = await JSONFilePreset('db.json', defaultData)
+const shops = db.data.shops;
+
+export async function getAvailableId() {
+  const ids = shops.map(shop => shop.id);
+  const availableId = Math.max(...ids) + 1;
+  return availableId;
+}
+
+export async function createShop(req, res) {
+  const id = getAvailableId();
+  const address = {
+    city: req.body.location.city,
+    address: req.body.location.address
+  }
+  const name = req.body.name;
+  const phoneNumber = req.body.phonenumber;
+  const openingHours = {
+    monday: req.body.openingHours.monday,
+    tuesday: req.body.openingHours.tuesday,
+    wednesday: req.body.openingHours.wednesday,
+    thursday: req.body.openingHours.thursday,
+    friday: req.body.openingHours.friday,
+    saturday: req.body.openingHours.saturday,
+    sunday: req.body.openingHours.sunday
+  };
+  const payingMethods = req.body.payingMethods;
+  const userID = req.body.userID;
+
+  if (!name || !address.city || !address.address || !userID) {
```

Comment:
Check all required fields, like opening hours, because these do not have a check